src/Akismet.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Akismet;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Http\Message\ResponseInterface;
+
+class Akismet
+{
+    public $isConfigured = true;
+    private $apiUrl;
+    private $flarumVersion;
+    private $extensionVersion;
+
+    private $params = [];
+    public $proTip;
+
+    public function __construct(string $apiKey, string $homeUrl, $flarumVersion, $extensionVersion, $inDebugMode = false)
+    {
+        if (empty($apiKey)) {
+            return $this->isConfigured = false;
+        }
```

Comment:
Why not `$this->isConfigured = !empty($apiKey)`? Alternatively, a method could be used if we added the api key as a private property.